sale_coupon_advanced/models/sale_order.py

```diff
@@ -1,11 +1,66 @@
 # Copyright 2020 Camptocamp SA
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
-from odoo import models
+
+from odoo import _, models
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    def _get_reward_line_values(self, program):
+        """ Add the rewarded product if a reward line has been found for this
+            product
+        """
+        reward_values = super()._get_reward_line_values(program)
+        reward_product = program.reward_product_id
+        discount_product = program.discount_line_product_id
+        for reward_value in reward_values:
+            product_id = reward_value["product_id"]
+            is_reward = reward_value["is_reward_line"]
+            if product_id == discount_product.id and is_reward:
+                taxes = reward_product.taxes_id
+                if self.fiscal_position_id:
+                    taxes = self.fiscal_position_id.map_tax(taxes)
+                reward_values += [
+                    {
+                        "sequence": 998,  # Put this line before discount
+                        "name": reward_product.name,
+                        "product_id": reward_product.id,
+                        "price_unit": reward_product.lst_price,
+                        "is_reward_line": False,
```

Comment:
we may have it with "True", in this case, it will be removed when the program rollbacked
otherwise pls add to readme section about this feature, and mention it leads to not revertable changes, cos lines added with is_reward_line = False will not be removed